lib/daru/clean/clean.rb

```diff
@@ -0,0 +1,76 @@
+module Daru
+  # Module encapsulating advanced data cleaning methods on DataFrame.
+  module Clean
+    # replace elements with those from given set of elements
+    # based on similarity
+    # @param vector [Daru::Vector] vector to be cleansed
+    # @param dict [Array] the array serving as key set
+    # @example
+    #   df = Daru::DataFrame.new({ a: ['mal','femal','M','F','Fem']})
+    #   df.fuzzy_match :a, ['MALE','FEMALE']
+    #   # => #<Daru::DataFrame(5x1)>
+    #   #           a
+    #   #   0    MALE
+    #   #   1  FEMALE
+    #   #   2    MALE
+    #   #   3  FEMALE
+    #   #   4  FEMALE
+    def fuzzy_match vector, dict
+      # TODO
+    end
+
+    # fill missing places using linear regression
```

Comment:
1. Reason for this exact name?
2. I believe that there could be other means of filling missing places (like, IDK, mean of neighbour data; constant value; rolling fill that is already implemented; and so on). Don't you think this task in fact requires something like `fill(&algorythm)` + library of filling algorithms?

Reply:
1. The name was inspired by its use in the R package [imputeR](https://cran.r-project.org/web/packages/imputeR/imputeR.pdf)
2. Yes, we can add an interpolation module to statsample which will take care of the proposed generalized approach of filling algorithms. WDYT?